frontend/src/methods/labels.ts

```diff
@@ -75,3 +78,36 @@ export const selectors = (labels: Label[]) => {
     return `${label.key}=${label.value}`;
   });
 }
+
+const labelDescriptions = {
+  "invalid-state": "The machine is expected to be unallocated, but still has the configuration of a cluster.\nIt might be required to wipe the machine bypassing Omni."
+};
+
+export const getLabelFromID = (key: string, value: string): Label => {
+  const isUser = key.indexOf(SystemLabelPrefix) !== 0;
+
+  let strippedKey = key.replace(new RegExp(`^${SystemLabelPrefix}`), "");
+  let icon: string | undefined;
+  let color = getLabelColor(strippedKey);
+  let description = labelDescriptions[strippedKey];
+
+  if (strippedKey.indexOf("infra-provider") === 0) {
```

Comment:
No big deal but should we instead have this as constant on backend and use `tsgen:` and use the var both on backend and frontend?

Reply:
Good point!